src/docs/sphinx/blueprint_o2mrelation.rst

```diff
@@ -0,0 +1,123 @@
+
+Protocol
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To conform to the **o2mrelation** protocol, a *Node* must have the following characteristics:
+
+ * Be of an *Object* type (*List* types are not allowed)
+ * Contain at least one child that is a numeric leaf
+
+The numeric leaf/leaves of the *Node* must not be under any of the following paths, which all have special meanings and particular requirements when specified as part of an **o2mrelation**:
+
+ * ``sizes``: An integer leaf that specifies the number "many" items associated with each "one" in the relationship.
+ * ``offsets``: An integer leaf that denotes the start index of the "many" sequence for each "one" in the relationship.
+ * ``indices``: An integer leaf that indicates the index values of items in the values array(s).
+
+All of the above paths are optional and will resolve to simple defaults if left unspecified. These defaults are outlined below:
+
+ * ``sizes``: An array of ones (i.e. ``[1, 1, 1, ...]``) to indicate that the values have one-to-one correspondance.
+ * ``offsets``: An array of monotonitcally increasing index values (i.e. ``[0, 1, 2, ...]``) to indicate that the values are compacted.
+ * ``indices``: An array of monotonitcally increasing index values (i.e. ``[0, 1, 2, ...]``) to indicate that the values are ordered sequentially.
+
+Taken in sum, the consituents of the **o2mrelation** schema describe how data (contained in numeric leaves and indexed through ``indices``) maps in grouped clusters (defined by ``sizes`` and ``offsets``) from a source space (the "one" space) to a destination space (the "many" space).
+
+.. note::
+   While the ``sizes``, ``offsets``, and ``indices`` meta-components of the **o2mrelation** definition are
+   independently defined, they interplay in ways that aren't immediately obvious. The most commonly missed
+   of these "gotcha" behaviors are defined below:
+
+   * Every **o2mrelation** must define both or neither of ``sizes`` and ``offsets``.
+   * If none of the meta-component paths are specified, their defaults set the **o2mrelation** to be a compacted, one-to-one relationship.
+   * The ``sizes`` and ``offsets`` values always refer to entries in ``indices``. If ``indices`` isn't present, it defaults to a "pass through" index, so in this case ``sizes`` and ``offsets`` can be thought of as indexing directly into the numeric leaves.
+
+Properties, Queries, and Transforms
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+ * **conduit::blueprint::o2mrelation::query_paths(const Node &o2mrelation, Node &res)**
+
+     Fills in the ``res`` node with detected value paths found in ``o2mrelation``.
+
+     * Example: {values: [int64], sizes: [int64], offsets: [int32], meta: [char8]} => [values]
+
+ * **conduit::blueprint::o2mrelation::to_compact(Node &o2mrelation)**
+
+     Updates the contents of the ``offsets`` array so that it refers to a compacted sequence of relationships.
+
+     * Example: sizes: {2, 5, 3, 8} => offsets: {0, 2, 7, 10}
```

Comment:
Yes, that's precisely the intent! This function will fill the place of `conduit::blueprint::mesh::topology::unstructured::generate_offsets` in covering this use case. 

This new function is currently a tad clumsy because it requires that the given node be a conformal `o2mrelation` (just like all other `conduit::blueprint::(namespace)::query/transform` functions), which means the user will have to do something like `o2mrelation["offsets"].set(...)` prior to calling. This isn't all bad because it forces the user be very explicit about the data type they want, but it does add some complexity.

Reply:
Thanks.  I have no doubt that lots of data / algorithms out there only have the sizes.